spec/memoize_query/cached_select/bind_params_spec.rb

```diff
@@ -0,0 +1,91 @@
+RSpec.describe RedisMemo::MemoizeQuery::CachedSelect::BindParams do
+  it 'unions' do
+    left = described_class.new
+    left.params[1] << { a: 1 }
+
+    right = described_class.new
+    right.params[1] << { b: 2 }
+
+    bp = left.union(right)
+    bp.extract!
+    expect(bp.params[1].to_a).to eq([{ a: 1 }, { b: 2 }])
+
+    left.params[1] << { b: 3 }
+    bp.extract!
+    expect(bp.params[1].to_a).to eq([{ a: 1 }, { b: 3 }, { b: 2 }])
+  end
+
+  it 'products' do
```

Comment:
I think a more complicated test case with a query that contains both a product and union would be beneficial